model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -204,73 +212,70 @@ std::unique_ptr<ResultBase> SegmentationModel::postprocess(InferenceResult& infR
-            cv::Mat predictions(outHeight, outWidth, CV_32SC1);
-            const auto data = outTensor.data<int64_t>();
-            for (size_t i = 0; i < predictions.total(); ++i) {
-                reinterpret_cast<int32_t*>(predictions.data)[i] = int32_t(data[i]);
-            }
-            predictions.convertTo(result->resultImage, CV_8UC1);
-        } else if (outTensor.get_element_type() == ov::element::f32) {
-            const float* data = outTensor.data<float>();
-            for (int rowId = 0; rowId < outHeight; ++rowId) {
-                for (int colId = 0; colId < outWidth; ++colId) {
-                    int classId = 0;
-                    float maxProb = -1.0f;
-                    for (int chId = 0; chId < outChannels; ++chId) {
-                        float prob = data[chId * outHeight * outWidth + rowId * outWidth + colId];
-                        if (prob > maxProb) {
-                            classId = chId;
-                            maxProb = prob;
-                        }
-                    }  // nChannels
-
-                    result->resultImage.at<uint8_t>(rowId, colId) = classId;
-                }  // width
-            }  // height
+    cv::Mat soft_prediction;
+    if (outChannels == 1 && outTensor.get_element_type() == ov::element::i32) {
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1, outTensor.data<int32_t>());
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outChannels == 1 && outTensor.get_element_type() == ov::element::i64) {
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1);
+        const auto data = outTensor.data<int64_t>();
+        for (size_t i = 0; i < predictions.total(); ++i) {
+            reinterpret_cast<int32_t*>(predictions.data)[i] = int32_t(data[i]);
         }
-
-        cv::resize(result->resultImage,
-                result->resultImage,
-                cv::Size(inputImgSize.inputImgWidth, inputImgSize.inputImgHeight),
-                0,
-                0,
-                cv::INTER_NEAREST);
-
-        return std::unique_ptr<ResultBase>(result);
-    }
-    float* data = outTensor.data<float>();
-    std::vector<cv::Mat> channels;
-    for (size_t c = 0; c < outTensor.get_shape()[1]; ++c) {
-        channels.emplace_back(cv::Size{outHeight, outWidth}, CV_32F, data + c * outHeight * outWidth);
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outTensor.get_element_type() == ov::element::f32) {
+        float* data = outTensor.data<float>();
+        std::vector<cv::Mat> channels;
+        for (size_t c = 0; c < outTensor.get_shape()[1]; ++c) {
+            channels.emplace_back(cv::Size{outWidth, outHeight}, CV_32FC1, data + c * outHeight * outWidth);
+        }
+        cv::merge(channels, soft_prediction);
+        cv::resize(soft_prediction, soft_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
```

Comment:
You've changed the algo. Originally it was:
1. blur
2. resize

now it is
1. resize
2. blur

Is this intentional?

Reply:
I agree that resize must be applied sooner or later. The question is only about the order

Reply:
Got confirmation that it's blur first, resize later. Will apply.